Out-of-source build issue with pxdgen

In this setup (the Gentoo build process), the build dir is below the source dir. And our buildsystem doesn't handle this properly.

* source-dir: `/tmp/portage/games-strategy/openage-9999/work/openage-9999`
* bin-dir: `/tmp/portage/games-strategy/openage-9999/work/openage-9999_build`

```
% make pxdgen VERBOSE=1
/usr/bin/cmake -S/tmp/portage/games-strategy/openage-9999/work/openage-9999 -B/tmp/portage/games-strategy/openage-9999/work/openage-9999_build --check-build-system CMakeFiles/Makefile.cmake 0
make -f CMakeFiles/Makefile2 pxdgen
make[1]: Entering directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
/usr/bin/cmake -S/tmp/portage/games-strategy/openage-9999/work/openage-9999 -B/tmp/portage/games-strategy/openage-9999/work/openage-9999_build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /tmp/portage/games-strategy/openage-9999/work/openage-9999_build/CMakeFiles 1
make -f CMakeFiles/Makefile2 CMakeFiles/pxdgen.dir/all
make[2]: Entering directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
make -f CMakeFiles/pxdgen.dir/build.make CMakeFiles/pxdgen.dir/depend
make[3]: Entering directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
cd /tmp/portage/games-strategy/openage-9999/work/openage-9999_build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/portage/games-strategy/openage-9999/work/openage-9999 /tmp/portage/games-strategy/openage-9999/work/openage-9999 /tmp/portage/games-strategy/openage-9999/work/openage-9999_build /tmp/portage/games-strategy/openage-9999/work/openage-9999_build /tmp/portage/games-strategy/openage-9999/work/openage-9999_build/CMakeFiles/pxdgen.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
make -f CMakeFiles/pxdgen.dir/build.make CMakeFiles/pxdgen.dir/build
make[3]: Entering directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
[100%] pxdgen: generating .pxd files from headers
cd /tmp/portage/games-strategy/openage-9999/work/openage-9999 && /usr/bin/python3.6 -m buildsystem.pxdgen --file-list /tmp/portage/games-strategy/openage-9999/work/openage-9999_build/py/pxdgen_sources --output-dir /tmp/portage/games-strategy/openage-9999/work/openage-9999_build
Traceback (most recent call last):
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/portage/games-strategy/openage-9999/work/openage-9999/buildsystem/pxdgen.py", line 479, in <module>
    main()
  File "/tmp/portage/games-strategy/openage-9999/work/openage-9999/buildsystem/pxdgen.py", line 473, in main
    initfile.relative_to(CWD)))
  File "/usr/lib64/python3.6/pathlib.py", line 874, in relative_to
    .format(str(self), str(formatted)))
ValueError: '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build/libopenage/__init__.py' does not start with '/tmp/portage/games-strategy/openage-9999/work/openage-9999'
make[3]: *** [CMakeFiles/pxdgen.dir/build.make:94: py/pxdgen_timefile] Error 1
make[3]: Leaving directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
make[2]: *** [CMakeFiles/Makefile2:1198: CMakeFiles/pxdgen.dir/all] Error 2
make[2]: Leaving directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
make[1]: *** [CMakeFiles/Makefile2:1209: CMakeFiles/pxdgen.dir/rule] Error 2
make[1]: Leaving directory '/tmp/portage/games-strategy/openage-9999/work/openage-9999_build'
make: *** [Makefile:652: pxdgen] Error 2
```

--- a/buildsystem/pxdgen.py
+++ b/buildsystem/pxdgen.py
@@ -470,7 +470,7 @@ def main():
                 initfile = template.with_suffix("." + extension)
                 if not initfile.exists():
                     print("\x1b[36mpxdgen: create package index %s\x1b[0m" % (
-                        initfile.relative_to(CWD)))
+                        initfile.relative_to(args.output_dir)))
 
                     initfile.touch()
 

--- a/buildsystem/python.cmake
+++ b/buildsystem/python.cmake
@@ -328,6 +328,10 @@ function(python_finalize)
 	write_on_change("${CMAKE_BINARY_DIR}/py/pxd_list" "${pxd_list};${generated_pxd_list}")
 	set(CYTHONIZE_TIMEFILE "${CMAKE_BINARY_DIR}/py/cythonize_timefile")
 	add_custom_command(OUTPUT "${CYTHONIZE_TIMEFILE}"
+		# remove unneeded files from the previous builds (if any)
+		COMMAND "${CMAKE_COMMAND}" -E remove -f
+			"${CMAKE_BINARY_DIR}/__init__.py"
+			"${CMAKE_BINARY_DIR}/__init__.pxd"
 		COMMAND "${PYTHON}" -m buildsystem.cythonize
 		"${CMAKE_BINARY_DIR}/py/cython_modules"
 		"${CMAKE_BINARY_DIR}/py/cython_modules_embed"

